lib/synapse/config_generator/haproxy.rb

```diff
@@ -816,19 +816,24 @@ def initialize(opts)
       @opts['do_socket'] = true unless @opts.key?('do_socket')
       @opts['do_checks'] = false unless @opts.key?('do_checks')
       @opts['do_reloads'] = true unless @opts.key?('do_reloads')
-      req_pairs = {
-        'do_writes' => 'config_file_path',
-        'do_socket' => 'socket_file_path',
-        'do_checks' => 'check_command',
-        'do_reloads' => 'reload_command'
+      req_groups = {
+        'do_writes' => ['config_file_path'],
+        'do_socket' => ['socket_file_path'],
+        'do_checks' => ['check_command', 'candidate_config_file_path'],
+        'do_reloads' => ['reload_command'],
       }
 
-      req_pairs.each do |cond, req|
-        if @opts[cond]
-          raise ArgumentError, "the `#{req}` option is required when `#{cond}` is true" unless @opts[req]
+      req_groups.each do |cond, reqs|
+        reqs.each do |req_config|
+          if @opts[cond]
+            raise ArgumentError, "the `#{req_config}` option is required when `#{cond}` is true" unless @opts[req_config]
```

Comment:
also worth sending metrics before crashing?  so ppl will know that crashing is due to invalid configuration

Reply:
Discussed in-person; Synapse already has crash metrics globally: https://github.com/airbnb/synapse/blob/master/lib/synapse.rb#L36. This will output a metric for any exception.

Reply:
Discussed in-person; Synapse already has crash metrics globally: https://github.com/airbnb/synapse/blob/master/lib/synapse.rb#L36. This will output a metric for any exception.